components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -313,6 +315,61 @@ struct SwapResponse {
   // Note we could also expose "sources" later
 };
 
+// Structs to model Jupiter (swap) HTTP API interactions
+// Swagger spec: https://quote-api.jup.ag/docs/static/index.html
+//
+// [XXX] - amount field is incorrectly modelled as number in spec.
+struct JupiterFee {
+  uint64 amount;
+  string mint;
+  double pct;
+};
+
+struct JupiterMarketInfo {
+  string id;
+  string label;
+  string input_mint;
+  string output_mint;
+  bool not_enough_liquidity;
+  uint64 in_amount;
+  uint64 out_amount;
+  double price_impact_pct;
+  JupiterFee lp_fee;
+  JupiterFee platform_fee;
+};
+
+struct JupiterRoute {
+  uint64 in_amount;
+  uint64 out_amount;
+  uint64 amount;
+  uint64 other_amount_threshold;
+  string swap_mode;
+  double price_impact_pct;
+  array<JupiterMarketInfo> market_infos;
+};
+
+struct JupiterSwapQuote {
+  array<JupiterRoute> routes;
+};
+
+struct JupiterQuoteParams {
+  string input_mint;
+  string output_mint;
+  string amount;
+  double slippage_percentage;
```

Comment:
Yes and yes. Other optional parameters are:
- `swapMode` => default value is `ExactIn` and is the only one available on Solana mainnet.
- `feeBps` => static value, so supplied by the clients.
- `userPublicKey` => for now, I skipped this since it is mainly to obtain detailed fees. Might add this in a different PR if we feel the need.
- `onlyDirectRoutes` => default value gives the best quotes, so no specific need to change that from the client.